lib/http-bridge/test/integration/Cardano/WalletSpec.hs

```diff
@@ -60,17 +69,28 @@ spec = do
                 (WalletName "My Wallet")
                 (mkSeqState @(HttpBridge 'Testnet) (xprv, mempty) minBound)
             unsafeRunExceptT $ W.restoreWallet wallet wid
-            threadDelay 2000000
-            tip <- slotId . currentTip . fst <$>
-                unsafeRunExceptT (W.readWallet wallet wid)
-            unless (tip > slotMinBound) $
-                expectationFailure ("The wallet tip is still " ++ show tip)
+
+            let policy = limitRetriesByCumulativeDelay
+                    (60 * second)
+                    (constantDelay (1 * second))
+            let shouldRetry _ = \case
+                    Left _ -> pure True
+                    Right _ -> pure False
+            let assertion _ = do
+                    tip <- slotId . currentTip . fst <$>
+                        unsafeRunExceptT (W.readWallet wallet wid)
+                    return $ if tip > slotMinBound
+                        then Right ()
+                        else Left ("The wallet tip is still " <> show tip)
+            result <- retrying policy shouldRetry assertion
+            when (isLeft result) $ expectationFailure (show result)
   where
-    port = 1337
+    second = 1000*1000
     closeBridge (handle, _) = do
         cancel handle
-        threadDelay 1000000
+        threadDelay second
```

Comment:
This might not be needed anymore maybe because of the dynamic `findPort`

Reply:
indeed, but I am paranoid.